pkgs/development/rocm-modules/6/clr/default.nix

```diff
@@ -100,6 +100,16 @@ in stdenv.mkDerivation (finalAttrs: {
       url = "https://github.com/ROCm/clr/commit/77c581a3ebd47b5e2908973b70adea66891159ee.patch";
       hash = "sha256-auBedbd7rghlKav7A9V6l64J7VmtE9GizIdi5gWj+fs=";
     })
+    (fetchpatch {
+      name = "Improve-hipamd-compat-check.patch";
+      url = "https://github.com/GZGavinZhao/clr/commit/f52172a0767f88bf386dc615a3354156d023bdb8.patch";
+      hash = "sha256-kbEeJsQgAxbNfBCPB2Jny4z526UAnIxHNQLGsD2iFvg=";
+    })
+    (fetchpatch {
+      name = "improve-rocclr-isa-compatibility-check.patch";
```

Comment:
> Where is this from?

I wrote the patch for Debian.
https://salsa.debian.org/rocm-team/rocm-hipamd/-/blob/debian/5.7.1-3/debian/patches/0025-improve-rocclr-isa-compatibility-check.patch

> Are we allowed to include this?

Yes.

Reply:
Thank you @cgmb! @mschwaig to prevent confusion, I've now switched the patches' url to Debian whenever possible. The only ones that are not switched are those that don't apply cleanly and needs manual fixes/rebase.

Reply:
This looks good.

For the ´rocclr isa compatibility check´, are you aware that the new patch is slightly different?

Just asking to make sure this is intended.

```diff
1d0
< From 2783c57b0f225ad8bc553e2d244837d57d8375bc Mon Sep 17 00:00:00 2001
4c3
< Subject: [PATCH] improve rocclr isa compatibility check
---
> Subject: improve rocclr isa compatibility check
17,18d15
< 
< Signed-off-by: Gavin Zhao <git@gzgz.dev>
20,21c17,18
<  rocclr/device/device.cpp | 39 ++++++++++++++++++++++++++++++++++++---
<  1 file changed, 36 insertions(+), 3 deletions(-)
---
>  rocclr/device/device.cpp | 45 ++++++++++++++++++++++++++++++++++++++++++---
>  1 file changed, 42 insertions(+), 3 deletions(-)
23,24d19
< diff --git a/rocclr/device/device.cpp b/rocclr/device/device.cpp
< index 0249f31d6..739bb027c 100644
27c22
< @@ -234,10 +234,43 @@ std::string Isa::isaName() const {
---
> @@ -232,10 +232,49 @@
49,51c44,47
< +        const std::array<uint32_t, 4> equivalent_gfx90x = { 0, 2, 9, 12 };
< +        if (Contains(equivalent_gfx90x, codeObjectIsa.versionStepping()) &&
< +            Contains(equivalent_gfx90x, agentIsa.versionStepping())) {
---
> +      const std::array<uint32_t, 4> gfx900_equivalent = { 0, 2, 9, 12 };
> +      const std::array<uint32_t, 5> gfx900_superset = { 0, 2, 6, 9, 12 };
> +      if (Contains(gfx900_equivalent, codeObjectIsa.versionStepping()) &&
> +          Contains(gfx900_superset, agentIsa.versionStepping())) {
55a52,55
> +        const std::array<uint32_t, 1> gfx1010_equivalent = { 0 };
> +        const std::array<uint32_t, 4> gfx1010_superset = { 0, 1, 2, 3 };
> +        if (Contains(gfx1010_equivalent, codeObjectIsa.versionStepping()) &&
> +            Contains(gfx1010_superset, agentIsa.versionStepping())) {
56a57
> +        }
```

Reply:
Yes, I'm aware. The new patch is just being more cautious on compatibility by checking both the code object ISA and the host ISA. The difference is intended.